lib/queue.js

```diff
@@ -0,0 +1,118 @@
+'use strict'
+
+var async = require('async')
+var chalk = require('chalk')
+var debug = require('debug-log')('mashape-analytics-queue')
+var http = require('http')
+var pkg = require('../package.json')
+var util = require('util')
+
+/**
+ * ALF Queue
+ */
+var Queue = module.exports = function (serviceToken, environment, options) {
+  var self = this
+
+  this.opts = options
+  this.timeout = null
+  this.userAgent = util.format('%s/%s', 'mashape-analytics-agent-node', pkg.version)
+
+  // setup event queue
+  // TODO specify worker pool
+  // TODO use msgpack + gzip?
+  this.queue = async.queue(function (entry, done) {
+    // append entry to log
+    self.alf.har.log.entries.push(entry)
+
+    // start timer
+    if (!self.timeout) {
+      self.timeout = setTimeout(function () {
```

Comment:
I'm more advocating *separation of function*, let the queue handle queuing and a timer handle timeouts ... you're doing both in one now.

also, what's bad in this flow is the timer is initialized *after* adding to the queue, meaning, the timer wont ever reset until the NEXT time a request comes in and AFTER the entry was added to the queue ...

this is why I say the timer should be outside the queue logic, its independent of it. every `x` seconds flush whatever is in the queue ... and start a new one, even re-initiating a new queue object.